torchvision/prototype/transforms/functional/_temporal.py

```diff
@@ -0,0 +1,24 @@
+import torch
+
+from torchvision.prototype import features
+
+
+def uniform_temporal_subsample(
+    inpt: features.VideoTypeJIT, num_samples: int, temporal_dim: int = -4
+) -> features.VideoTypeJIT:
+    if isinstance(inpt, torch.Tensor) and (torch.jit.is_scripting() or not isinstance(inpt, features.Video)):
+        return uniform_temporal_subsample_video(inpt, num_samples, temporal_dim=temporal_dim)
+    else:  # isinstance(inpt, features.Video)
+        if temporal_dim != -4:
+            raise ValueError("Video inputs must have temporal_dim equal to -4")
+        output = uniform_temporal_subsample_video(
+            inpt.as_subclass(torch.Tensor), num_samples, temporal_dim=temporal_dim
+        )
+        return features.Video.wrap_like(inpt, output)
+
+
+def uniform_temporal_subsample_video(video: torch.Tensor, num_samples: int, temporal_dim: int = -4) -> torch.Tensor:
+    # Reference: https://github.com/facebookresearch/pytorchvideo/blob/a0a131e/pytorchvideo/transforms/functional.py#L19
+    t_max = video.size(temporal_dim) - 1
```

Comment:
`Tensor.size()` is a somewhat outdated method. Plus, we don't use it anywhere so far. Just use `shape`

```suggestion
    t_max = video.shape[temporal_dim] - 1
```